presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcPageSource.java

```diff
@@ -74,23 +80,33 @@
     private int batchId;
     private long completedPositions;
     private boolean closed;
+    private boolean rowsDeleteLoad;
+    private Optional<BitSet> rowsDeleted = Optional.empty();
 
     public OrcPageSource(
+            OrcStorageManager orcStorageManager,
```

Comment:
I don't think we need to pass `OrcStorageManager` and `FileSystem` to `OrcPageSource`. Alternatively, we can pass a function which accepts a `UUID` and returns a `BitSet` (i.e. `uuid -> getRowsFromUuid(filesystem, uuid)`).